### ISSUE TITLE
[Rust] Conflict between `PartialEq` and `ChunkCompare` traits

### ISSUE THREAD

**Issue by ibENPC** [pola-rs/polars#1816]
### Problem
Suppose (😉) that we want to implement the [PartialEq](https://doc.rust-lang.org/std/cmp/trait.PartialEq.html) trait for the `Series` structure:
```rust
impl PartialEq for Series {
    fn eq(&self, other: &Self) -> bool {
        // Some stuff
    }
}
```
Then the compiler would complain for example like this:
![tmp](https://user-images.githubusercontent.com/72026000/142420541-37e16d95-5ffd-46cc-8aec-2dededd9b052.png)

That is because the trait **`ChunkCompare`**, implemented for a `Series`, **is taking the name for the function `eq`**.

### Potential solution
- Rename the `eq` function in the `ChunkCompare` trait. The new name might be `chunk_eq`.

### Why bother?
I think that it is better to be aligned with the Rust standards. It easily improves the ergonomic for the developer and the code quality for the long term.

If this conflict is resolved, I can make a PR that implements the `PartialEq` for the `Series` structure.

**Comment by ritchie46** [comment]
Now I remember why we didn't implement that. ;)

Maybe we can refactor the `ChunkCompare` trait to longer names. E.g. `equal`, `not_equal`

**Comment by ibENPC** [comment]
Solved in #1826
